server-info-db-worker: Invalid column family specified in write batch

run tinkerpop structure tests:
```
2020-07-02 16:47:46 267883 [server-info-db-worker-1] [ERROR] com.baidu.hugegraph.backend.Transaction [] - Failed to commit changes:
com.baidu.hugegraph.backend.BackendException: Exception in backend
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:682) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:1) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStore.commitTx(RocksDBStore.java:490) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitMutation2Backend(AbstractTransaction.java:295) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.commit2Backend(IndexableTransaction.java:57) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commit(AbstractTransaction.java:210) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.commit(GraphTransaction.java:491) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:345) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$2(ServerInfoManager.java:209) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:308) [classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: org.rocksdb.RocksDBException: Invalid column family specified in write batch
	at org.rocksdb.RocksDB.write0(Native Method) ~[rocksdbjni-6.8.1.jar:?]
	at org.rocksdb.RocksDB.write(RocksDB.java:1586) ~[rocksdbjni-6.8.1.jar:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:679) ~[classes/:?]
	... 15 more
```

```
2020-07-02 16:47:46 267896 [task-db-worker-1] [ERROR] com.baidu.hugegraph.backend.Transaction [] - Failed to commit changes:
com.baidu.hugegraph.backend.BackendException: Exception in backend
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:682) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:1) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStore.commitTx(RocksDBStore.java:490) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitMutation2Backend(AbstractTransaction.java:295) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.commit2Backend(IndexableTransaction.java:57) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commit(AbstractTransaction.java:210) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.commit(GraphTransaction.java:491) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:345) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573) ~[classes/:?]
	at com.baidu.hugegraph.task.StandardTaskScheduler.lambda$1(StandardTaskScheduler.java:418) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:308) [classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: org.rocksdb.RocksDBException: Invalid column family specified in write batch
	at org.rocksdb.RocksDB.write0(Native Method) ~[rocksdbjni-6.8.1.jar:?]
	at org.rocksdb.RocksDB.write(RocksDB.java:1586) ~[rocksdbjni-6.8.1.jar:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.commit(RocksDBStdSessions.java:679) ~[classes/:?]
	... 15 more
```

```
2020-07-02 16:47:46 267833 [task-scheduler-1] [ERROR] com.baidu.hugegraph.task.TaskManager [] - Exception occurred when schedule job
com.baidu.hugegraph.HugeException: Failed to update/query server info: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:272) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.save(ServerInfoManager.java:199) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.heartbeat(ServerInfoManager.java:171) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager.scheduleOrExecuteJob(TaskManager.java:258) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_111]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_111]
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:270) ~[classes/:?]
	... 10 more
Caused by: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:358) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$2(ServerInfoManager.java:209) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:308) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_111]
	... 3 more
```

memory:
```
2020-07-02 19:59:25 1037884 [task-worker-1] [ERROR] com.baidu.hugegraph.backend.Transaction [] - Failed to remove vertices
java.lang.IllegalStateException: Graph 'standardhugegraph[hugegraph_structure_standard]' has been closed
	at com.google.common.base.Preconditions.checkState(Preconditions.java:531) ~[guava-25.1-jre.jar:?]
	at com.baidu.hugegraph.util.E.checkState(E.java:68) ~[hugegraph-common-1.7.8.jar:1.7.8.0]
	at com.baidu.hugegraph.StandardHugeGraph.checkGraphNotClosed(StandardHugeGraph.java:310) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.schemaTransaction(StandardHugeGraph.java:329) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.existsLinkLabel(StandardHugeGraph.java:578) ~[classes/:?]
	at com.baidu.hugegraph.schema.VertexLabel.existsLinkLabel(VertexLabel.java:74) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.prepareDeletions(GraphTransaction.java:353) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.prepareCommit(GraphTransaction.java:296) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.commit2Backend(IndexableTransaction.java:53) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commit(AbstractTransaction.java:210) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.commit(GraphTransaction.java:491) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.traverseByLabel(GraphTransaction.java:1883) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.traverseVerticesByLabel(GraphTransaction.java:1842) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.removeVertices(GraphTransaction.java:1795) [classes/:?]
	at com.baidu.hugegraph.job.schema.VertexLabelRemoveCallable.removeVertexLabel(VertexLabelRemoveCallable.java:83) [classes/:?]
	at com.baidu.hugegraph.job.schema.VertexLabelRemoveCallable.execute(VertexLabelRemoveCallable.java:45) [classes/:?]
	at com.baidu.hugegraph.job.SysJob.call(SysJob.java:29) [classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at com.baidu.hugegraph.task.HugeTask.run(HugeTask.java:269) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
2020-07-02 19:59:25 1037888 [task-worker-1] [WARN ] com.baidu.hugegraph.task.HugeTask [] - An exception occurred when running task: 26174
com.baidu.hugegraph.HugeException: Failed to remove vertices
	at com.baidu.hugegraph.backend.tx.GraphTransaction.removeVertices(GraphTransaction.java:1802) ~[classes/:?]
	at com.baidu.hugegraph.job.schema.VertexLabelRemoveCallable.removeVertexLabel(VertexLabelRemoveCallable.java:83) ~[classes/:?]
	at com.baidu.hugegraph.job.schema.VertexLabelRemoveCallable.execute(VertexLabelRemoveCallable.java:45) ~[classes/:?]
	at com.baidu.hugegraph.job.SysJob.call(SysJob.java:29) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at com.baidu.hugegraph.task.HugeTask.run(HugeTask.java:269) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: Graph 'standardhugegraph[hugegraph_structure_standard]' has been closed
	at com.google.common.base.Preconditions.checkState(Preconditions.java:531) ~[guava-25.1-jre.jar:?]
	at com.baidu.hugegraph.util.E.checkState(E.java:68) ~[hugegraph-common-1.7.8.jar:1.7.8.0]
	at com.baidu.hugegraph.StandardHugeGraph.checkGraphNotClosed(StandardHugeGraph.java:310) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.schemaTransaction(StandardHugeGraph.java:329) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.existsLinkLabel(StandardHugeGraph.java:578) ~[classes/:?]
	at com.baidu.hugegraph.schema.VertexLabel.existsLinkLabel(VertexLabel.java:74) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.prepareDeletions(GraphTransaction.java:353) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.prepareCommit(GraphTransaction.java:296) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.commit2Backend(IndexableTransaction.java:53) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commit(AbstractTransaction.java:210) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.commit(GraphTransaction.java:491) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.traverseByLabel(GraphTransaction.java:1883) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.traverseVerticesByLabel(GraphTransaction.java:1842) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.removeVertices(GraphTransaction.java:1795) ~[classes/:?]
	... 10 more
```

```
2020-07-03 23:09:08 24420 [main] [INFO ] com.baidu.hugegraph.HugeGraph [] - Init server info for graph 'hugegraph_structure_standard'...
2020-07-03 23:09:32 47789 [server-info-db-worker-1] [ERROR] com.baidu.hugegraph.backend.Transaction [] - Failed to commit changes:
com.baidu.hugegraph.backend.BackendException: Table 'hugegraph_structure_standard+vi' is not opened
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions.cf(RocksDBStdSessions.java:277) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions.access$3(RocksDBStdSessions.java:274) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStdSessions$StdSession.put(RocksDBStdSessions.java:661) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBTable.insert(RocksDBTable.java:95) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBTable.append(RocksDBTable.java:114) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStore.mutate(RocksDBStore.java:370) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBStore.mutate(RocksDBStore.java:353) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitMutation2Backend(AbstractTransaction.java:293) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.commit2Backend(IndexableTransaction.java:57) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commit(AbstractTransaction.java:210) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.commit(GraphTransaction.java:491) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:345) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$2(ServerInfoManager.java:272) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:306) [classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
2020-07-03 23:09:51 67164 [task-scheduler-1] [ERROR] com.baidu.hugegraph.task.TaskManager [] - Exception occurred when schedule job
com.baidu.hugegraph.HugeException: Failed to update/query server info: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Table 'hugegraph_structure_standard+vi' is not opened
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:362) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.save(ServerInfoManager.java:262) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.heartbeat(ServerInfoManager.java:185) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager.scheduleOrExecuteJob(TaskManager.java:256) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Table 'hugegraph_structure_standard+vi' is not opened
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_111]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_111]
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:360) ~[classes/:?]
	... 10 more
Caused by: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Table 'hugegraph_structure_standard+vi' is not opened
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:358) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$2(ServerInfoManager.java:272) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:306) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_111]
	... 3 more
```

core test with rocksdb:
```
com.baidu.hugegraph.HugeException: Failed to update/query server info: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Undefined vertex label: '~server'
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:362)
	at com.baidu.hugegraph.task.ServerInfoManager.serverInfos(ServerInfoManager.java:328)
	at com.baidu.hugegraph.task.ServerInfoManager.serverInfos(ServerInfoManager.java:323)
	at com.baidu.hugegraph.task.ServerInfoManager.initServerInfo(ServerInfoManager.java:142)
	at com.baidu.hugegraph.StandardHugeGraph.serverStarted(StandardHugeGraph.java:204)
	at com.baidu.hugegraph.core.MultiGraphsTest.testCopySchemaWithMultiGraphs(MultiGraphsTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Undefined vertex label: '~server'
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:360)
	... 38 more
Caused by: java.lang.IllegalArgumentException: Undefined vertex label: '~server'
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:163)
	at com.baidu.hugegraph.util.E.checkArgument(E.java:56)
	at com.baidu.hugegraph.StandardHugeGraph.vertexLabel(StandardHugeGraph.java:567)
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$5(ServerInfoManager.java:335)
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:306)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

```
java.lang.AssertionError: com.baidu.hugegraph.HugeException: Failed to update/query server info: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Table 'schema_g1+v' is not opened
	at org.junit.Assert.fail(Assert.java:88)
	at com.baidu.hugegraph.core.MultiGraphsTest.destoryGraphs(MultiGraphsTest.java:344)
	at com.baidu.hugegraph.core.MultiGraphsTest.testCopySchemaWithMultiGraphsWithConflict(MultiGraphsTest.java:209)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)


```

tinkerpop test with rocksdb 6.10.2:
```
org.junit.AssumptionViolatedException: got: <true>, expected: is <false>
	at org.junit.Assume.assumeThat(Assume.java:95)
	at org.apache.tinkerpop.gremlin.structure.FeatureSupportTest$GraphVariablesFunctionalityTest.shouldEnableFeatureOnGraphIfNotEnabled(FeatureSupportTest.java:694)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:229)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:51)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at com.baidu.hugegraph.tinkerpop.StructureBasicSuite$1.evaluate(StructureBasicSuite.java:118)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite$1.evaluate(AbstractGremlinSuite.java:238)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

com.baidu.hugegraph.HugeException: Failed to update/query server info: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:307)
	at com.baidu.hugegraph.task.ServerInfoManager.save(ServerInfoManager.java:255)
	at com.baidu.hugegraph.task.ServerInfoManager.initServerInfo(ServerInfoManager.java:159)
	at com.baidu.hugegraph.StandardHugeGraph.serverStarted(StandardHugeGraph.java:204)
	at com.baidu.hugegraph.StandardHugeGraph.truncateBackend(StandardHugeGraph.java:269)
	at com.baidu.hugegraph.tinkerpop.TestGraph.truncateBackend(TestGraph.java:126)
	at com.baidu.hugegraph.tinkerpop.TestGraph.clearAll(TestGraph.java:111)
	at com.baidu.hugegraph.tinkerpop.TestGraphProvider.clear(TestGraphProvider.java:352)
	at org.apache.tinkerpop.gremlin.GraphManager$ManagedGraphProvider.clear(GraphManager.java:142)
	at org.apache.tinkerpop.gremlin.AbstractGremlinTest.tearDown(AbstractGremlinTest.java:151)
	at sun.reflect.GeneratedMethodAccessor7.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:229)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite.runChild(AbstractGremlinSuite.java:51)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at com.baidu.hugegraph.tinkerpop.StructureBasicSuite$1.evaluate(StructureBasicSuite.java:118)
	at org.apache.tinkerpop.gremlin.AbstractGremlinSuite$1.evaluate(AbstractGremlinSuite.java:238)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.util.concurrent.ExecutionException: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.baidu.hugegraph.task.ServerInfoManager.call(ServerInfoManager.java:305)
	... 60 more
Caused by: com.baidu.hugegraph.backend.BackendException: Failed to commit changes: Exception in backend
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.commitOrRollback(AbstractTransaction.java:358)
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.afterWrite(AbstractTransaction.java:312)
	at com.baidu.hugegraph.backend.tx.GraphTransaction.addVertex(GraphTransaction.java:573)
	at com.baidu.hugegraph.task.ServerInfoManager.lambda$2(ServerInfoManager.java:265)
	at com.baidu.hugegraph.task.TaskManager$ContextCallable.call(TaskManager.java:311)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
@@ -230,15 +230,17 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSchemaStore().open(this.configuration);
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
+
+        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+
         try {
-            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
             this.storeProvider.init();
             this.storeProvider.initSystemInfo(this);
         } finally {
+            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
-            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
         }
 
         LOG.info("Graph '{}' has been initialized", this.name);
@@ -251,14 +253,16 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSchemaStore().open(this.configuration);
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
+
+        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+
         try {
-            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
             this.storeProvider.clear();
         } finally {
+            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
-            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
         }
 
         LOG.info("Graph '{}' has been cleared", this.name);
@@ -268,14 +272,15 @@ public class StandardHugeGraph implements HugeGraph {
     public void truncateBackend() {
         this.waitUntilAllTasksCompleted();
 
+        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+
         try {
-            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
             this.storeProvider.truncate();
             this.storeProvider.initSystemInfo(this);
             this.serverStarted(this.serverInfoManager().selfServerId(),
                                this.serverInfoManager().selfServerRole());
         } finally {
-            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
+            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
         }
         LOG.info("Graph '{}' has been truncated", this.name);
     }

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
@@ -231,12 +231,14 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
         try {
+            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
             this.storeProvider.init();
             this.storeProvider.initSystemInfo(this);
         } finally {
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
+            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
         }
 
         LOG.info("Graph '{}' has been initialized", this.name);
@@ -250,11 +252,13 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
         try {
+            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
             this.storeProvider.clear();
         } finally {
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
+            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
         }
 
         LOG.info("Graph '{}' has been cleared", this.name);
@@ -264,11 +268,15 @@ public class StandardHugeGraph implements HugeGraph {
     public void truncateBackend() {
         this.waitUntilAllTasksCompleted();
 
-        this.storeProvider.truncate();
-        this.storeProvider.initSystemInfo(this);
-        this.serverStarted(this.serverInfoManager().selfServerId(),
-                           this.serverInfoManager().selfServerRole());
-
+        try {
+            LockUtil.lock(this.name, TaskManager.HEARTBEAT);
+            this.storeProvider.truncate();
+            this.storeProvider.initSystemInfo(this);
+            this.serverStarted(this.serverInfoManager().selfServerId(),
+                               this.serverInfoManager().selfServerRole());
+        } finally {
+            LockUtil.unlock(this.name, TaskManager.HEARTBEAT);
+        }
         LOG.info("Graph '{}' has been truncated", this.name);
     }
 

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
@@ -231,13 +231,13 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
 
-        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+        LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
 
         try {
             this.storeProvider.init();
             this.storeProvider.initSystemInfo(this);
         } finally {
-            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
+            LockUtil.unlock(this.name, LockUtil.GRAPH_LOCK);
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
@@ -254,12 +254,12 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadSystemStore().open(this.configuration);
         this.loadGraphStore().open(this.configuration);
 
-        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+        LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
 
         try {
             this.storeProvider.clear();
         } finally {
-            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
+            LockUtil.unlock(this.name, LockUtil.GRAPH_LOCK);
             this.loadGraphStore().close();
             this.loadSystemStore().close();
             this.loadSchemaStore().close();
@@ -272,7 +272,7 @@ public class StandardHugeGraph implements HugeGraph {
     public void truncateBackend() {
         this.waitUntilAllTasksCompleted();
 
-        LockUtil.lock(this.name, TaskManager.GRAPH_LOCK);
+        LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
 
         try {
             this.storeProvider.truncate();
@@ -280,7 +280,7 @@ public class StandardHugeGraph implements HugeGraph {
             this.serverStarted(this.serverInfoManager().selfServerId(),
                                this.serverInfoManager().selfServerRole());
         } finally {
-            LockUtil.unlock(this.name, TaskManager.GRAPH_LOCK);
+            LockUtil.unlock(this.name, LockUtil.GRAPH_LOCK);
         }
         LOG.info("Graph '{}' has been truncated", this.name);
     }

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/StandardHugeGraph.java
@@ -232,7 +232,6 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadGraphStore().open(this.configuration);
 
         LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
-
         try {
             this.storeProvider.init();
             this.storeProvider.initSystemInfo(this);
@@ -255,7 +254,6 @@ public class StandardHugeGraph implements HugeGraph {
         this.loadGraphStore().open(this.configuration);
 
         LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
-
         try {
             this.storeProvider.clear();
         } finally {
@@ -273,7 +271,6 @@ public class StandardHugeGraph implements HugeGraph {
         this.waitUntilAllTasksCompleted();
 
         LockUtil.lock(this.name, LockUtil.GRAPH_LOCK);
-
         try {
             this.storeProvider.truncate();
             this.storeProvider.initSystemInfo(this);
@@ -282,6 +279,7 @@ public class StandardHugeGraph implements HugeGraph {
         } finally {
             LockUtil.unlock(this.name, LockUtil.GRAPH_LOCK);
         }
+
         LOG.info("Graph '{}' has been truncated", this.name);
     }
 

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/ServerInfoManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/ServerInfoManager.java
@@ -200,6 +200,11 @@ public class ServerInfoManager {
         return 10000;
     }
 
+    protected boolean initialized() {
+        return this.call(() -> this.graph.graph().backendStoreSystemInfo()
+                                         .exists());
+    }
+
     protected synchronized HugeServerInfo pickWorkerNode(
                                           Collection<HugeServerInfo> servers,
                                           HugeTask<?> task) {
@@ -325,8 +330,7 @@ public class ServerInfoManager {
     }
 
     private HugeServerInfo removeSelfServerInfo() {
-        if (this.call(() -> this.graph.graph()
-                                .backendStoreSystemInfo().exists())) {
+        if (this.initialized()) {
             return this.removeServerInfo(this.selfServerId);
         }
         return null;

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/StandardTaskScheduler.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/StandardTaskScheduler.java
@@ -116,6 +116,10 @@ public class StandardTaskScheduler implements TaskScheduler {
         return this.graph.graph();
     }
 
+    public String graphName() {
+        return this.graph.name();
+    }
+
     @Override
     public int pendingTasks() {
         return this.tasks.size();

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -258,33 +258,36 @@ public final class TaskManager {
         try {
             for (TaskScheduler entry : this.schedulers.values()) {
                 StandardTaskScheduler scheduler = (StandardTaskScheduler) entry;
-                ServerInfoManager server = scheduler.serverManager();
+                ServerInfoManager serverManager = scheduler.serverManager();
 
                 String graph = scheduler.graph().name();
                 LockUtil.lock(graph, LockUtil.GRAPH_LOCK);
                 graphs.add(graph);
 
                 // Skip if not initialized(maybe truncated or cleared)
-                if (!server.initialized()) {
+                if (!serverManager.initialized()) {
                     continue;
                 }
 
                 // Update server heartbeat
-                server.heartbeat();
+                serverManager.heartbeat();
 
                 /*
                  * Master schedule tasks to suitable servers.
                  * There is no suitable server when these tasks are created
                  */
-                if (server.master()) {
+                if (serverManager.master()) {
                     scheduler.scheduleTasks();
+                    if (!serverManager.onlySingleNode()) {
+                        continue;
+                    }
                 }
 
                 // Schedule queued tasks scheduled to current server
-                scheduler.executeTasksOnWorker(server.selfServerId());
+                scheduler.executeTasksOnWorker(serverManager.selfServerId());
 
                 // Cancel tasks scheduled to current server
-                scheduler.cancelTasksOnWorker(server.selfServerId());
+                scheduler.cancelTasksOnWorker(serverManager.selfServerId());
             }
         } catch (Throwable e) {
             LOG.error("Exception occurred when schedule job", e);

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -45,7 +45,7 @@ public final class TaskManager {
 
     private static final Logger LOG = Log.logger(TaskManager.class);
 
-    public static final String HEARTBEAT = "heartbeat";
+    public static final String GRAPH_LOCK = "graph_lock";
 
     public static final String TASK_WORKER = "task-worker-%d";
     public static final String TASK_DB_WORKER = "task-db-worker-%d";
@@ -263,7 +263,7 @@ public final class TaskManager {
                 ServerInfoManager server = scheduler.serverManager();
 
                 String graph = scheduler.graph().name();
-                LockUtil.lock(graph, HEARTBEAT);
+                LockUtil.lock(graph, GRAPH_LOCK);
                 graphs.add(graph);
 
                 // Skip if not initialized(maybe truncated or cleared)
@@ -293,7 +293,7 @@ public final class TaskManager {
         } finally {
             Collections.reverse(graphs);
             for (String graph : graphs) {
-                LockUtil.unlock(graph, HEARTBEAT);
+                LockUtil.unlock(graph, GRAPH_LOCK);
             }
         }
     }

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -95,16 +95,21 @@ public final class TaskManager {
     }
 
     public void closeScheduler(HugeGraphParams graph) {
-        TaskScheduler scheduler = this.schedulers.get(graph);
-        if (scheduler != null && scheduler.close()) {
-            this.schedulers.remove(graph);
-        }
-        if (!this.taskExecutor.isTerminated()) {
-            this.closeTaskTx(graph);
-        }
+        LockUtil.lock(graph.name(), LockUtil.GRAPH_LOCK);
+        try {
+            TaskScheduler scheduler = this.schedulers.get(graph);
+            if (scheduler != null && scheduler.close()) {
+                this.schedulers.remove(graph);
+            }
+            if (!this.taskExecutor.isTerminated()) {
+                this.closeTaskTx(graph);
+            }
 
-        if (!this.schedulerExecutor.isTerminated()) {
-            this.closeSchedulerTx(graph);
+            if (!this.schedulerExecutor.isTerminated()) {
+                this.closeSchedulerTx(graph);
+            }
+        } finally {
+            LockUtil.unlock(graph.name(), LockUtil.GRAPH_LOCK);
         }
     }
 
@@ -260,7 +265,7 @@ public final class TaskManager {
                 StandardTaskScheduler scheduler = (StandardTaskScheduler) entry;
                 ServerInfoManager serverManager = scheduler.serverManager();
 
-                String graph = scheduler.graph().name();
+                String graph = scheduler.graphName();
                 LockUtil.lock(graph, LockUtil.GRAPH_LOCK);
                 graphs.add(graph);
 

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -20,6 +20,7 @@
 package com.baidu.hugegraph.task;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Queue;
@@ -37,11 +38,14 @@ import com.baidu.hugegraph.HugeException;
 import com.baidu.hugegraph.HugeGraphParams;
 import com.baidu.hugegraph.util.E;
 import com.baidu.hugegraph.util.ExecutorUtil;
+import com.baidu.hugegraph.util.LockUtil;
 import com.baidu.hugegraph.util.Log;
 
 public final class TaskManager {
 
     private static final Logger LOG = Log.logger(TaskManager.class);
+
+    public static final String HEARTBEAT = "heartbeat";
 
     public static final String TASK_WORKER = "task-worker-%d";
     public static final String TASK_DB_WORKER = "task-db-worker-%d";
@@ -252,10 +256,20 @@ public final class TaskManager {
     }
 
     private void scheduleOrExecuteJob() {
+        List<String> graphs = new ArrayList<>();
         try {
             for (TaskScheduler entry : this.schedulers.values()) {
                 StandardTaskScheduler scheduler = (StandardTaskScheduler) entry;
                 ServerInfoManager server = scheduler.serverManager();
+
+                String graph = scheduler.graph().name();
+                LockUtil.lock(graph, HEARTBEAT);
+                graphs.add(graph);
+
+                // Skip if not initialized(maybe truncated or cleared)
+                if (!server.initialized()) {
+                    continue;
+                }
 
                 // Update server heartbeat
                 server.heartbeat();
@@ -276,6 +290,11 @@ public final class TaskManager {
             }
         } catch (Throwable e) {
             LOG.error("Exception occurred when schedule job", e);
+        } finally {
+            Collections.reverse(graphs);
+            for (String graph : graphs) {
+                LockUtil.unlock(graph, HEARTBEAT);
+            }
         }
     }
 

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -94,22 +94,17 @@ public final class TaskManager {
         this.schedulers.put(graph, scheduler);
     }
 
-    public void closeScheduler(HugeGraphParams graph) {
-        LockUtil.lock(graph.name(), LockUtil.GRAPH_LOCK);
-        try {
-            TaskScheduler scheduler = this.schedulers.get(graph);
-            if (scheduler != null && scheduler.close()) {
-                this.schedulers.remove(graph);
-            }
-            if (!this.taskExecutor.isTerminated()) {
-                this.closeTaskTx(graph);
-            }
+    public synchronized void closeScheduler(HugeGraphParams graph) {
+        TaskScheduler scheduler = this.schedulers.get(graph);
+        if (scheduler != null && scheduler.close()) {
+            this.schedulers.remove(graph);
+        }
+        if (!this.taskExecutor.isTerminated()) {
+            this.closeTaskTx(graph);
+        }
 
-            if (!this.schedulerExecutor.isTerminated()) {
-                this.closeSchedulerTx(graph);
-            }
-        } finally {
-            LockUtil.unlock(graph.name(), LockUtil.GRAPH_LOCK);
+        if (!this.schedulerExecutor.isTerminated()) {
+            this.closeSchedulerTx(graph);
         }
     }
 
@@ -258,7 +253,7 @@ public final class TaskManager {
         }
     }
 
-    private void scheduleOrExecuteJob() {
+    private synchronized void scheduleOrExecuteJob() {
         List<String> graphs = new ArrayList<>();
         try {
             for (TaskScheduler entry : this.schedulers.values()) {

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -94,6 +94,15 @@ public final class TaskManager {
         this.schedulers.put(graph, scheduler);
     }
 
+    /*
+     * 'closeScheduler' should sync with 'scheduleOrExecuteJob'. Because
+     * 'closeScheduler' will be called by 'graph.close()' in main thread and
+     * there is gap between 'scheduler.close()'(will close graph tx) and
+     * 'this.schedulers.remove(graph)'. In this gap 'scheduleOrExecuteJob'
+     * may be run in scheduler-db-thread and 'scheduleOrExecuteJob' will
+     * reopen graph tx. As a result, graph tx will mistakenly not be closed
+     * after 'graph.close()'
+     */
     public synchronized void closeScheduler(HugeGraphParams graph) {
         TaskScheduler scheduler = this.schedulers.get(graph);
         if (scheduler != null && scheduler.close()) {

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskManager.java
@@ -45,8 +45,6 @@ public final class TaskManager {
 
     private static final Logger LOG = Log.logger(TaskManager.class);
 
-    public static final String GRAPH_LOCK = "graph_lock";
-
     public static final String TASK_WORKER = "task-worker-%d";
     public static final String TASK_DB_WORKER = "task-db-worker-%d";
     public static final String SERVER_INFO_DB_WORKER =
@@ -263,7 +261,7 @@ public final class TaskManager {
                 ServerInfoManager server = scheduler.serverManager();
 
                 String graph = scheduler.graph().name();
-                LockUtil.lock(graph, GRAPH_LOCK);
+                LockUtil.lock(graph, LockUtil.GRAPH_LOCK);
                 graphs.add(graph);
 
                 // Skip if not initialized(maybe truncated or cleared)
@@ -293,7 +291,7 @@ public final class TaskManager {
         } finally {
             Collections.reverse(graphs);
             for (String graph : graphs) {
-                LockUtil.unlock(graph, GRAPH_LOCK);
+                LockUtil.unlock(graph, LockUtil.GRAPH_LOCK);
             }
         }
     }

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
@@ -59,6 +59,7 @@ public final class LockUtil {
     public static final String PROPERTY_KEY_ADD_UPDATE = "pk_add_update";
     public static final String KEY_LOCK = "key_lock";
     public static final String ROW_LOCK = "row_lock";
+    public static final String REENTRANT_LOCK = "reentrant_lock";
 
     public static final long WRITE_WAIT_TIMEOUT = 30L;
 
@@ -160,6 +161,16 @@ public final class LockUtil {
         RowLock<K> rowLock = LockManager.instance().get(join(graph, ROW_LOCK))
                                         .rowLock(group);
         rowLock.unlockAll(rows);
+    }
+
+    public static void lock(String graph, String name) {
+        LockManager.instance().get(join(graph, REENTRANT_LOCK))
+                   .lock(name).lock();
+    }
+
+    public static void unlock(String graph, String name) {
+        LockManager.instance().get(join(graph, REENTRANT_LOCK))
+                   .lock(name).unlock();
     }
 
     public static List<Lock> lock(String... locks) {

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
@@ -74,6 +74,7 @@ public final class LockUtil {
         LockManager.instance().create(join(graph, PROPERTY_KEY_ADD_UPDATE));
         LockManager.instance().create(join(graph, KEY_LOCK));
         LockManager.instance().create(join(graph, ROW_LOCK));
+        LockManager.instance().create(join(graph, REENTRANT_LOCK));
     }
 
     public static void destroy(String graph) {
@@ -87,6 +88,7 @@ public final class LockUtil {
         LockManager.instance().destroy(join(graph, PROPERTY_KEY_ADD_UPDATE));
         LockManager.instance().destroy(join(graph, KEY_LOCK));
         LockManager.instance().destroy(join(graph, ROW_LOCK));
+        LockManager.instance().destroy(join(graph, REENTRANT_LOCK));
     }
 
     private static String join(String graph, String group) {

--- a/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
+++ b/hugegraph-core/src/main/java/com/baidu/hugegraph/util/LockUtil.java
@@ -61,6 +61,8 @@ public final class LockUtil {
     public static final String ROW_LOCK = "row_lock";
     public static final String REENTRANT_LOCK = "reentrant_lock";
 
+    public static final String GRAPH_LOCK = "graph_lock";
+
     public static final long WRITE_WAIT_TIMEOUT = 30L;
 
     public static void init(String graph) {

